### ISSUE TITLE
I found another potential vulnerability in the setPirexEth() function.

### ISSUE THREAD

**Issue by c4-bot-6** [code-423n4/redacted-bug-bounty#62]
# Lines of code

https://github.com/redacted-cartel/pirex-eth-contracts/blob/11f30c7e35b67d45deefe405c22a30f352bc5b21/src/AutoPxEth.sol#L194

# Vulnerability details

I found another potential vulnerability in the `setPirexEth()` function.

**Issue:**
The `setPirexEth()` function is used to set the PirexEth contract address, which is a critical component of the Smart Contract. However, the function does not include any checks to prevent the PirexEth contract address from being set to the zero address (address(0)). This can be potentially exploited by an attacker who gains control of the contract ownership, as setting the PirexEth contract address to the zero address would effectively disable the Smart Contract's functionality, leading to a denial of service.

**Impact:**
The impact of this vulnerability is significant, as it can lead to a denial of service for the Smart Contract. If an attacker gains control of the contract ownership, they can set the PirexEth contract address to the zero address, effectively disabling the Smart Contract's functionality. This can result in a significant loss of functionality and value for users, especially if the attacker is able to maintain control for a prolonged period of time.

**Recommendation:**
To mitigate this vulnerability, I recommend adding a check in the `setPirexEth()` function to prevent the PirexEth contract address from being set to the zero address. This can be achieved by adding the following line of code:

`require(_pirexEth != address(0), "PirexEth contract address cannot be set to the zero address.");`

This will ensure that the PirexEth contract address cannot be set to the zero address, preventing the denial of service for the Smart Contract.

**Disclosure:**
I recommend disclosing this vulnerability to the contract owner as soon as possible, so that they can take appropriate action to mitigate the risk. It is important to ensure that the vulnerability is addressed before it can be exploited by an attacker.

**Conclusion:**
In conclusion, the Smart Contract provided has a vulnerability in the `setPirexEth()` function that can potentially be exploited by an attacker to gain control of the contract ownership and set the PirexEth contract address to the zero address, leading to a denial of service for the Smart Contract. I recommend adding a check in the `setPirexEth()` function to prevent the PirexEth contract address from being set to the zero address, and disclosing the vulnerability to the contract owner as soon as possible.

I hope this report helps in securing the Smart Contract and preventing any potential attacks.
